sources/rest_api/README.md

```diff
@@ -0,0 +1,222 @@
+
+## Anatomy of the config object
+
+> **_TIP:_**  Import `RESTAPIConfig` from the `rest_api` module to have convenient tips.
+
+The config object passed to the REST API Generic Source has three main elements:
+
+```python
+my_config: RESTAPIConfig = {
+    "client": {
+        ...
+    },
+    "resource_defaults": {
+        ...
+    },
+    "resources": {
+        ...
+    ,
+}
+```
+
+
+
+`client` contains the configuration to connect to the APIs endpoints (e.g. base URL, authentication method, default behaviour for the paginator, and more).
+
+`resource_defaults` contains the default values to configure the dlt resources returned by this source.
+
+`resources` object contains the configuration for each resource. 
+
+The configuration with smallers scope will overwrite the one with the wider one:
+
+    Resource Configuration > Resource Defaults Configuration > Client Configuration
+
+## Reference
+
+### `client`
+
+#### `auth` [optional]
+Use the auth property to pass a token or a `HTTPBasicAuth` object for more complex authentication methods. Here some practical examples:
+
+1. Simple token (read from the `.dlt/secrets.toml` file):
+```python
+my_api_config: RESTAPIConfig = {
+    "client": {
+        "base_url": "https://my_api.com/api/v1/",
+        "auth": {
+            "token": dlt.secrets["sources.my_api.access_token"],
+        },
+    },
+    ...
+}
+```
+
+2. 
+```python
+from requests.auth import HTTPBasicAuth
+
+basic_auth = HTTPBasicAuth(dlt.secrets["sources.my_api.access_token"], "")
+
+my_api_config: RESTAPIConfig = {
+    "client": {
+        "base_url": "https://my_api.com/api/v1/",
+        "auth": basic_auth,
+    },
+    ...
+}
+```
+
+#### `base_url`
+The base URL that will be prepended to the endpoints specified in the `resources` objects. Example
+
+```python
+    "base_url": "https://my_api.com/api/v1/",
+```
+
+#### `paginator` [optional]
+The paginator property specify the default paginator to be used for the endpoint responses.
+
+//TODO describe paginators, the possible strings, and the disctionary to be used 
+
+Possible paginators are:
+- BasePaginator - 
+- HeaderLinkPaginator - String alias `"header_links"` -
+- JSONResponsePaginator -  String alias `"json_links"` - The pagination metainformation are in a node of the JSON response. 
+
+    Usage example for a response with the url of the next page located at `paging.next`:
+    ```python
+    "paginator": JSONResponsePaginator(
+                        next_key=["paging", "next"]
+                    )
+    ```
+- SinglePagePaginator -  String alias `"single_page"` - The response will be interepreted as a single page response, ignoring possible pagination metadata.
+- UnspecifiedPaginator - String alias `"auto"` - 
+
+
+#### `request_client` [optional]
+This property allows to passa a custom `requests` client.

```

Comment:
For the first release we decided to allow to pass only a custom Session object (not the dlt Client).
```suggestion
```

Reply:
Is this parameter still available or replace by something else (see Willi's comment)?